### PR TITLE
Cy create language

### PR DESCRIPTION
**Pull request by jemayn**
Remaking https://github.com/umbraco/Umbraco-CMS/pull/11307 to get around a GH bug that showed way more changes.

@madsrasmussen If I add the TS definitions like they are in tests/Umbraco.Tests.AcceptanceTest/cypress/typings.d.ts I get intellisense and it all works 🙂

We could use that alongside adding the commands to the commands.js file we get typed custom commands, which can then once in a while be pulled into @bergmania 's package.

### CODE REVIEW

File: tests/Umbraco.Tests.AcceptanceTest/cypress/integration/Languages/languages.ts [hunk 1]
```diff
@@ -1,10 +1,31 @@
...
 
     beforeEach(() => {
         cy.umbracoLogin(Cypress.env('username'), Cypress.env('password'), false);
       });
 
+    it('Creates language', () => {
+      // Setup
+      const language = 'Danish';
+      const culture = 'da';
+      cy.umbracoEnsureLanguageNotExists(culture);
```
Comment (Zeegaan) [inline tests/Umbraco.Tests.AcceptanceTest/cypress/integration/Languages/languages.ts:13]:
```suggestion
      cy.umbracoEnsureLanguageNameNotExists(culture);
```
Use the already existing language method

File: tests/Umbraco.Tests.AcceptanceTest/cypress/integration/Languages/languages.ts [hunk 2]
```diff
@@ -28,11 +49,11 @@ context('Languages', () => {
         cy.umbracoButtonByLabelKey('contentTypeEditor_yesDelete').click();
 
         // Assert there is only 2 language
-        cy.get('tbody > tr').should('have.length', 3);        
+        cy.get('tbody > tr').should('have.length', 2);        
 
         // Cleanup
         cy.umbracoEnsureLanguageNotExists(language1);
         cy.umbracoEnsureLanguageNotExists(language2);
```
Comment (Zeegaan) [inline tests/Umbraco.Tests.AcceptanceTest/cypress/integration/Languages/languages.ts:56]:
```suggestion
        cy.umbracoEnsureLanguageNameNotExists(language1);
        cy.umbracoEnsureLanguageNameNotExists(language2);
```

File: tests/Umbraco.Tests.AcceptanceTest/cypress/typings.d.ts [hunk 3]
```diff
@@ -3,3 +3,28 @@
...
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace Cypress {
+        interface Chainable<Subject> {
+            /**
+             * Checks to see if the language with specified name does not exist
+             * If it does it will automatically delete it
+             * @param  {string} name Name of language to delete
+             * @example cy.umbracoEnsureLanguageNotExists('Danish');
+             */
+            umbracoEnsureLanguageNotExists: (name: string) => Chainable<void>;
```
Comment (Zeegaan) [inline tests/Umbraco.Tests.AcceptanceTest/cypress/typings.d.ts:19]: As mentioned in your other PR we already have a method called cy.umbracoEnsureLanguageNameNotExists(name)
```suggestion
    namespace Cypress {
        interface Chainable<Subject> {
```

**Review by Zeegaan — CHANGES_REQUESTED** [review]
Found a few things that needs changing 👍

**Comment by jemayn** [comment]
@Zeegaan 

The existing command matches on the language name, not the culture like this one. I've updated the method on this PR to make it clearer that it's on culture 🙂

**Comment by elit0451** [comment]
Just FYI: The just merged PR https://github.com/umbraco/Umbraco-CMS/pull/11527 already moved the **languages.ts** to the `tests` folder, instead of `src`. 🙂 
So make sure to get the latest changes from `umbraco:v9/contrib`.

**Comment by mikecp** [comment]
Hello @jemayn ,

Thanks for this PR, it's a nice addition to the acceptance tests suite 😉👍

It all works good, but before merging, I wanted to check with you, in the line of @Zeegaan 's remarks earlier, if there was a specific reason why you needed to add the method  `umbracoEnsureLanguageCultureNotExists` and could not use the existing one `umbracoEnsureLanguageNameNotExists`.

I just want to make sure that we do not create what seems to be an "almost duplicate" method if there is no specific reason, but maybe I am overseeing something. 😁

Thanks in advance for your feedback!

**Comment by jemayn** [comment]
> Hello @jemayn ,
> 
> Thanks for this PR, it's a nice addition to the acceptance tests suite 😉👍
> 
> It all works good, but before merging, I wanted to check with you, in the line of @Zeegaan 's remarks earlier, if there was a specific reason why you needed to add the method `umbracoEnsureLanguageCultureNotExists` and could not use the existing one `umbracoEnsureLanguageNameNotExists`.
> 
> I just want to make sure that we do not create what seems to be an "almost duplicate" method if there is no specific reason, but maybe I am overseeing something. 😁
> 
> Thanks in advance for your feedback!

Hey @mikecp !

I can amend the PR to use the existing method if you want, but I felt this was a better method than the existing one - the existing one relies on the language names which can be anything, this method uses the culture which is still just a string - but feels a bit safer to use..

**Comment by mikecp** [comment]
Hi @jemayn ,

Thanks for clearing the difference out, looking on a culture name seems indeed safer, so let's keep that method in there 😉!

**Review by mikecp — APPROVED** [review]

**Pull request merged**